tests/sentry/release_health/test_metrics_sessions_v2.py

```diff
@@ -0,0 +1,99 @@
+
+from django.urls import reverse
+
+from sentry.release_health.duplex import compare_results
+from sentry.release_health.metrics import MetricsReleaseHealthBackend
+from sentry.sentry_metrics.indexer.mock import MockIndexer
+from sentry.testutils.cases import APITestCase, SessionMetricsTestCase, SnubaTestCase
+
+
+class MetricsSessionsV2SessionsTest(APITestCase, SnubaTestCase):
+    # Sessions cache. The sessions backend writes into it, and the metrics
+    # backend reads from it.
+    cache = {}
+
+    def setUp(self):
+        super().setUp()
+        self.setup_fixture()
+
+    def setup_fixture(self):
+        self.organization1 = self.organization
+        self.project1 = self.project
+        self.project2 = self.create_project(
+            name="teletubbies", slug="teletubbies", teams=[self.team], fire_project_created=True
+        )
+
+        self.release1 = self.create_release(project=self.project1, version="hello")
+        self.release2 = self.create_release(project=self.project1, version="hola")
+        self.release3 = self.create_release(project=self.project2, version="hallo")
+
+        self.environment1 = self.create_environment(self.project1, name="development")
+        self.environment2 = self.create_environment(self.project1, name="production")
+        self.environment3 = self.create_environment(self.project2, name="testing")
+
+    def do_request(self, query, user=None, org=None):
+        self.login_as(user=user or self.user)
+        url = reverse(
+            "sentry-api-0-organization-sessions",
+            kwargs={"organization_slug": (org or self.organization1).slug},
+        )
+        return self.client.get(url, query, format="json")
+
+    def get_sessions_data(self, groupby, interval):
+        response = self.do_request(
+            {
+                "organization_slug": [self.organization1],
+                "project": [self.project1.id],
+                "field": ["sum(session)"],
+                "groupBy": [groupby],
+                "interval": interval,
+            }
+        )
+        assert response.status_code == 200
+        return response.data
+
+    def test_sessions_metrics_equal_num_keys(self):
+        """
+        Tests whether the number of keys in the metrics implementation of
+        sessions data is the same as in the sessions implementation.
+
+        Runs twice. Firstly, against sessions implementation to populate the
+        cache. Then, against the metrics implementation, and compares with
+        cached results.
+
+        Note that this test only checks whether the number of keys in both
+        responses are equal, other errors (such as keys in different order)
+        may happen and we're fine with them.
+        """
+        empty_groupbyes = ["project", "release", "environment", "session.status"]
+        interval_days = "1d"
+
+        from sentry.api.endpoints.organization_sessions import release_health
+
+        if not release_health.is_metrics_based():
```

Comment:
You could get rid of this `if - else` and the `MetricsSessionsV2SessionsTest.cache` if you compared the response to a fixture rather than comparing it to the "sessions" response. So something like:

```python
for groupby, expected_groups in [
    (["project"], []),
    (["release"], []),
    (["environment"], []),
    (["session.status"], [<insert expected result here>]),
    (["release", "session.status"], []),
]:
    response = self.get_sessions_data(groupby)
    assert result_sorted(response.data)["groups"] == expected_groups
```

Reply:
I thought about that, but I decided to avoid it. My fear is to use that approach, change something in the implementation in any end, and have false positives or negatives without noticing. I think we will benefit more from it in a later stage of metrics.

What do you think?